cmd/podman/pods/create.go

```diff
@@ -105,29 +114,21 @@ func create(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	netInput, err := cmd.Flags().GetString("network")
-	if err != nil {
-		return err
-	}
-	n := specgen.Namespace{}
-	switch netInput {
-	case "bridge":
-		n.NSMode = specgen.Bridge
-	case "host":
-		n.NSMode = specgen.Host
-	case "slip4netns":
-		n.NSMode = specgen.Slirp
-	default:
-		if strings.HasPrefix(netInput, "container:") { // nolint
-			split := strings.Split(netInput, ":")
-			if len(split) != 2 {
-				return errors.Errorf("invalid network paramater: %q", netInput)
-			}
-			n.NSMode = specgen.FromContainer
-			n.Value = split[1]
-		} else if strings.HasPrefix(netInput, "ns:") {
-			return errors.New("the ns: network option is not supported for pods")
-		} else {
+	if cmd.Flag("network").Changed {
+		netInput, err := cmd.Flags().GetString("network")
+		if err != nil {
+			return err
+		}
+		n := specgen.Namespace{}
+		switch netInput {
+		case "bridge":
+			n.NSMode = specgen.Bridge
+		case "host":
+			n.NSMode = specgen.Host
+		case "slip4netns":
```

Comment:
Is this really "slip4netns" and not "slirp4netns"?  If the original is correct, could we change it?

Reply:
Good catch, fixed